File: solutions/java/cars-assemble/1/src/main/java/CarsAssemble.java
import java.lang.Exception;

public class CarsAssemble {

    public double productionRatePerHour(int speed) {
        double production = 221 * speed;

        switch (speed){
            case 5:
            case 6:
            case 7:
            case 8:
                production *= 0.9;
                break;
            case 9:
                production *= 0.8;
                break;
            case 10:
                production *= 0.77;
                break;
        }

        return production;
    }

    public int workingItemsPerMinute(int speed) {
        return (int)(this.productionRatePerHour(speed) / 60);
    }
}

File: solutions/csharp/cars-assemble/1/CarsAssemble.cs
using System;

static class AssemblyLine
{
    static int CARS_PER_HOUR = 221;
    
    public static double SuccessRate(int speed)
    {
        if (speed >= 1 && speed <= 4) {
            return 1.0;
        } else if (speed >= 5 && speed <= 8) {
            return 0.9;
        } else if (speed == 9) {
            return 0.8;
        } else if (speed == 10) {
            return 0.77;
        } else {
            return 0.0;
        }
    }
    
    public static double ProductionRatePerHour(int speed)
    {
        return AssemblyLine.CARS_PER_HOUR * speed * AssemblyLine.SuccessRate(speed);
    }

    public static int WorkingItemsPerMinute(int speed)
    {
        return (int)(AssemblyLine.ProductionRatePerHour(speed) / 60);
    }
}
